django/contrib/admin/options.py

```diff
@@ -2257,19 +2266,22 @@ def history_view(self, request, object_id, extra_context=None):
             .order_by("action_time")
         )
 
-        paginator = self.get_paginator(request, action_list, 100)
-        page_number = request.GET.get(PAGE_VAR, 1)
-        page_obj = paginator.get_page(page_number)
-        page_range = paginator.get_elided_page_range(page_obj.number)
+        Pagination = self.get_pagination(request)
+        pagination = Pagination(
```

Comment:
I feel like having both `Pagination` and `pagination` can be quite confusing, especially since we have `get_pagination` that returns the class and not the instance.

This would look better to me if we had:
- `pagination_class` and `get_pagination_class` instead of `Pagination` and `get_pagination`
- maybe also have a new `get_pagination` that would be in charge of calling `get_pagination_class` and instantiating the pagination class instead of directly calling the `Pagination` class.

This would be similar to a pattern we have in other places in Django, like in the `FormView` https://ccbv.co.uk/projects/Django/5.0/django.views.generic.edit/FormView/#get_form